src/Events/Controllers/SubscriptionController.cs

```diff
@@ -274,28 +279,44 @@ private static bool AuthorizeIdentityForConsumer(Subscription eventsSubscription
 
         /// <summary>
         /// Validates that the identity (user, organization or org) is authorized to create subscriptions for a given subject.
-        /// Currently the subject needs to match the identity.  Org does not need subject.
+        ///  Subscriptions created by org do not need subject.
         /// </summary>
         private async Task<bool> AuthorizeSubjectForConsumer(Subscription eventsSubscription)
         {
-            // First version require that created and consumer is the same
             if (eventsSubscription.CreatedBy.StartsWith(UserPrefix))
             {
                 int userId = Convert.ToInt32(eventsSubscription.CreatedBy.Replace(UserPrefix, string.Empty));
                 UserProfile profile = await _profileService.GetUserProfile(userId);
                 string ssn = PersonPrefix + profile.Party.SSN;
 
-                if (!ssn.Equals(eventsSubscription.AlternativeSubjectFilter))
+                if (ssn.Equals(eventsSubscription.AlternativeSubjectFilter))
                 {
-                    return false;
+                    return true;
+                }
+
+                bool hasRoleAccess = await _authorizationService.AuthorizeConsumerForEventsSubcription(eventsSubscription);
+
+                if (hasRoleAccess)
+                {
+                    return true;
                 }
             }
-            else if (!string.IsNullOrEmpty(eventsSubscription.SubjectFilter) && !eventsSubscription.SubjectFilter.Equals(eventsSubscription.Consumer))
+            else if (eventsSubscription.CreatedBy.StartsWith(OrgPrefix))
             {
-                return false;
+                if (string.IsNullOrEmpty(eventsSubscription.SubjectFilter))
```

Comment:
Will any application owner now be able to subscribe to events from any app (by any application owner)?

Reply:
My goal here was to not change the behavior of the authorization for app owners or organizations. Currently, an app owner can subscribe to any source as long as the subject isn't specified. Events will only be sent out if the consumer is authorized through the app policy.

We might need a larger discussion on what should be allowed for app owners and organisations. 

I think we need a larger discussion on what kinds of subscriptions that are relevant for app owners and organisations, and that this could be driven by the requirements from the app owners. Not sure if that needs to be bundled in with this change necessarily, but now we have a momentum for the discussion.